### ISSUE TITLE
SD card randomly disconnects while copying files in Win11

### ISSUE THREAD

**Issue by DJZSwag** [utmapp/UTM#4007]
**Describe the issue**
I am using an ARM Win11 VM, and I needed to copy a couple of files created by Wii U USB Helper to put on to my SD card. About a quarter or halfway through, File Explorer suddenly said something about how the target device was not available. When I clicked on the SD card, and it said that the device was not recognized. This is not because of Wii U USB Helper, because it worked fine on my Windows laptop. Also, for some reason, UTM thinks the SD card is a USB. 
**Configuration**
* UTM Version: Version 3.1.5 (53)
* OS Version: 12.4 Beta (21F5063e) (I will update soon)
* Intel or Apple Silicon? Apple M1

**Debug log**
What's even stranger is when I tried to do it again just for the log, the SD card didn't show up at all. It was there for a millisecond, then disappeared.

**Upload VM**
[config.plist.zip](https://github.com/utmapp/UTM/files/8680532/config.plist.zip)

P.S When is 3.2 coming out?

**Comment by JacksonChen666** [comment]
hardware connection issue?
make sure it is not that first of all. idk how, but just check on macOS?

**Comment by DJZSwag** [comment]
It works perfectly fine on macOS

**Comment by JacksonChen666** [comment]
checklist of things to check (or to more isolate this issue):
- [x] works on macOS
- [ ] does not behave the same on macOS (including any actions also done in windows, if possible)
- [ ] issue is persistent on windows VM (like it just stops working at a specific point every time)
- [x] issue does not appear on other computers
- [ ] issue is not persistent while using the same connector on macOS & another machine

**Comment by DJZSwag** [comment]
Works On macOS
Does not appear on other computers
The issue is persistent (yesterday it disconnected, a few hours ago it didn't even connect, but now it's fine)

**Comment by JacksonChen666** [comment]
> Works On macOS

no boxes were checked

> Does not appear on other computers

no boxes were checked

> The issue is persistent (yesterday it disconnected, a few hours ago it didn't even connect, but now it's fine)

that's not really the idea of persistent i was looking for, but i was really looking for *consistency*.

**Comment by conath** [comment]
@DJZSwag What model SD Card reader are you passing to the VM? Is it built in to your Mac or external? If external, is there a dock/dongle between the reader and your Mac?

**Comment by DJZSwag** [comment]
I'm using an HP DHC-CT203
![hp](https://user-images.githubusercontent.com/87071118/168239367-482a2a13-e27e-44d4-8e0c-c9c717fe16f7.jpeg)

**Comment by DJZSwag** [comment]
It's either I'm stupid or this is an actual bug. Sometimes when I plug my SD in, I don't plug it in all the way and thus doesn't appear on macOS. And the issue is REALLY inconsistent.

**Comment by DJZSwag** [comment]
It now works fine, and it's either because of 3.2.5, or it was because I didn't plus the SD all the way in.

**_And thus the story comes to an end,
with the help of no friend,
He committed user error,
and shall do it again never._**

**Comment by JacksonChen666** [comment]
(this is for your future reference, @DJZSwag. if you get anymore issues that are similar, reopen this issue or add a comment on the result of what's explained below)

i just thought of something that should help pin point if it's either a software bug (UTM) or hardware issue (connection issue?)

when you (@DJZSwag) use the VM and the SD card normally and it suddenly disconnects from the system, check the attached USB devices immediately in UTM

![attached USB devices in UTM menu in the menubar of a VM](https://user-images.githubusercontent.com/50755746/169077583-35527333-130e-4c7a-b24f-acef232dff10.png)

(that above is just an example, actual list varies)

when comparing before it disconnected and after it did, check if the desired device(s) is in the list. if the SD card is not in the list (before it did, now it doesn't), i would pin point it to a hardware issue (or a software issue at another level, the kernel/driver handling).

**Comment by DJZSwag** [comment]
Nope. All the times I plugged my SD card in, it read it as USB Storage (1:3), and the errors happened likely because I dont plug it in all the way.

And would you look at that! I was typing this while Windows was shutting down, and UTM crashed!

**Comment by osy** [comment]
Please attach a crash log.
